index.bs

```diff
@@ -2112,62 +2124,69 @@ partial interface MLGraphBuilder {
                 : {{MLInputOperandLayout/"nchw"}}
                 ::
                     1. Let |batches| be |inputShape|[0].
-                    1. Let |channels| be |inputShape|[1].
+                    1. Let |inputChannels| be |inputShape|[1].
                     1. Let |inputHeight| be |inputShape|[2].
                     1. Let |inputWidth| be |inputShape|[3].
                 : {{MLInputOperandLayout/"nhwc"}}
                 ::
                     1. Let |batches| be |inputShape|[0].
                     1. Let |inputHeight| be |inputShape|[1].
                     1. Let |inputWidth| be |inputShape|[2].
-                    1. Let |channels| be |inputShape|[3].
+                    1. Let |inputChannels| be |inputShape|[3].
             </dl>
         1. Let |filterShape| be |filter|'s [=MLOperand/shape=].
         1. Switch on |options|.{{MLConvTranspose2dOptions/filterLayout}}:
             <dl class=switch>
                 : {{MLConvTranspose2dFilterOperandLayout/"iohw"}}
                 ::
+                    1. Let |filterInputChannels| be |filterShape|[0].
+                    1. Let |filterOutputChannels| be |filterShape[1].
                     1. Let |filterHeight| be |filterShape|[2].
                     1. Let |filterWidth| be |filterShape|[3].
                 : {{MLConvTranspose2dFilterOperandLayout/"hwoi"}}
                 ::
                     1. Let |filterHeight| be |filterShape|[0].
                     1. Let |filterWidth| be |filterShape|[1].
+                    1. Let |filterOutputChannels| be |filterShape[2].
+                    1. Let |filterInputChannels| be |filterShape|[3].
                 : {{MLConvTranspose2dFilterOperandLayout/"ohwi"}}
                 ::
+                    1. Let |filterOutputChannels| be |filterShape[0].
                     1. Let |filterHeight| be |filterShape|[1].
                     1. Let |filterWidth| be |filterShape|[2].
+                    1. Let |filterInputChannels| be |filterShape|[3].
             </dl>
-        1. Let |outputSizes| be the result of [=MLGraphBuilder/calculating convtranspose2d output sizes=] given |inputHeight|, |inputWidth|, |filterHeight|, |filterWidth|, |options|.{{MLConvTranspose2dOptions/padding}}, |options|.{{MLConvTranspose2dOptions/strides}}, |options|.{{MLConvTranspose2dOptions/dilations}}, and |options|.{{MLConvTranspose2dOptions/outputPadding}}.
+        1. If |inputChannels| is not equal to |filterInputChannels|, then [=exception/throw=] a {{TypeError}}.
+        1. Let |outputChannels| be |filterOutputChannels| * |options|.{{MLConvTranspose2dOptions/groups}}
+        1. If |options|.{{MLConvTranspose2dOptions/bias}} [=map/exists=]:
+            1. If |options|.{{MLConvTranspose2dOptions/bias}}'s [=MLOperand/rank=] is not 1, then [=exception/throw=] a {{TypeError}}.
+            1. If |options|.{{MLConv2dOptions/bias}}'s [=MLOperand/shape=][0] is not equal to |outputChannels|, then [=exception/throw=] a {{TypeError}}.
```

Comment:
```suggestion
            1. If |options|.{{MLConvTranspose2dOptions/bias}}'s [=MLOperand/shape=][0] is not equal to |outputChannels|, then [=exception/throw=] a {{TypeError}}.
```

Reply:
Good catch! Fixed in https://github.com/webmachinelearning/webnn/pull/608/commits/2cf1f202ac66f287c5175bdb5ff922a8f95cea3f